ResearchSuite/ResearchSuite/RSDUIAction.swift

```diff
@@ -34,73 +34,168 @@
- */
+/// `RSDSkipToUIAction` implements an extension of the base protocol where the action includes an identifier for
+/// a step to skip to if this action is called.
 public protocol RSDSkipToUIAction : RSDUIAction {
     
-    /**
-     The identifier for the step to skip to if the action is called.
-     */
+    /// The identifier for the step to skip to if the action is called.
     var skipToIdentifier: String { get }
 }
 
 
-/**
- The action handler implements the custom actions of the step.
- */
+/// `RSDUIActionHandler` implements the custom actions of the step.
 public protocol RSDUIActionHandler {
     
-    /**
-     Customizable actions to return for a given action type. The `RSDStepController` can use these to customize the display of buttons to the user. If nil, `shouldHideAction()` will be called to determine if the default action should be used or if the action button should be hidden.
-     
-     @param actionType  The action type for the button.
-     @param step        The step that the action is on.
-     
-     @return            A custom UI action for this button. If nil, the default action will be used.
-     */
+    /// Customizable actions to return for a given action type. The `RSDStepController` can use these to
+    /// customize the display of buttons to the user. If nil, `shouldHideAction()` will be called to
+    /// determine if the default action should be used or if the action button should be hidden.
+    ///
+    /// - parameters:
+    ///     - actionType:  The action type for the button.
+    ///     - step:        The step that the action is on.
+    /// - returns: A custom UI action for this button. If nil, the default action will be used.
     func action(for actionType: RSDUIActionType, on step: RSDStep) -> RSDUIAction?
     
-    /**
-     Should the action button be hidden?
-     
-     @param actionType  The action type for the button.
-     @param step        The step that the action is on.
-     
-     @return            Whether or not the button should be hidden or `nil` if there is no explicit action.
-     */
+    /// Should the action button be hidden?
+    ///
+    /// - parameters:
+    ///     - actionType:  The action type for the button.
+    ///     - step:        The step that the action is on.
+    /// - returns: Whether or not the button should be hidden or `nil` if there is no explicit action.
     func shouldHideAction(for actionType: RSDUIActionType, on step: RSDStep) -> Bool?
 }
+
+
+/// The `RSDUIActionType` enum describes standard navigation actions that are common to a
+/// given UI step. It is extendable using the custom field.
+///
+public enum RSDUIActionType {
+    
+    /// Standard navigation elements that are common to most steps.
+    case navigation(Navigation)
+    
+    /// Standard navigation elements that are common to most steps.
```

Comment:
Should this just repeat the above comment?

Reply:
I have to document everything that is public or it doesn't get parsed.  :)